### PR TITLE
[BD-05] [BB-2992] [TNL-7578] Fix ORA grade explanation bug

### PR DESCRIPTION
**Pull request by aayushagra**
We found a bug where sometimes the grade explanation message on an ORA block would say that the grade is determined by the staff grade, even though staff grading was disabled on that particular block

**JIRA tickets**: TNL-7578

**Merge deadline**: None

**Testing instructions**:

1. Checkout this PR & install it on your local devstack
2. Make sure the grade explanation message now shows the correct response for all assessment types (Self Assessment, Peer Assessment, Staff Assessment)
3. Try overriding the grade using staff controls, and recheck that the grade explanation message now says Staff Grade, regardless of the assessment type

**Author notes and concerns**:

1. I am curious about why this bug occured on my master devstack but not the juniper devstack

**Reviewers**
- [ ] @0x29a

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @aayushagra! I've created [OSPR-5106](https://openedx.atlassian.net/browse/OSPR-5106) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: openassessment/xblock/openassessmentblock.py [hunk 1]
```diff
@@ -876,7 +875,13 @@ def valid_assessments(self):
 
     @property
     def assessment_steps(self):
-        return [asmnt['name'] for asmnt in self.valid_assessments]
+        assessment_steps = []
```
Comment (giovannicimolin) [inline openassessment/xblock/openassessmentblock.py:878]: Rename `asmnt` -> `assessment`

**Comment by natabene** [comment]
@aayushagra Thank you for your contribution. Please let me know once it is ready for our review.

**Comment by giovannicimolin** [comment]
@aayushagra Can you add `[BD-05]` to the title? Missing doing it makes the bot create the wrong ticket (`OSPR` instead of `BLENDED`). Let me know once this is ready for review again.

@natabene This is part of `[BD-05]` (`ORA Phase 1`) blended development project, sorry for the noise.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested this:
    1. Described bug can be reproduced (recent `edx-platrofm` and `edx-ora2`).
    2. Grading works correctly after installing ORA with changes from this PR>
- [x] I read through the code
- [ ] `assessment_steps` doesn't include documentation.

**Review by jnlapierre — APPROVED** [review]
LGTM , merge conflicts will need to be addressed.

**Comment by giovannicimolin** [comment]
@jnlapierre Merge conflicts fixed.

**Pull request merged**